### PR TITLE
test(backend): add xfail test for memtable with null strings

### PR DESCRIPTION
**Pull request by NickCrews**
The nulls are parsed into `"<NA>"`:

```
AssertionError: assert ['a', '<NA>', 'c'] == ['a', None, 'c']
```

run with

```
pytest ibis/backends/tests/test_generic.py -m duckdb -k test_memtable_null_str
```

### CODE REVIEW

File: ibis/backends/tests/test_generic.py [hunk 1]
```diff
@@ -916,6 +916,46 @@ def test_memtable_bool_column(backend, con, monkeypatch):
...
+)
+@pytest.mark.broken(
+    ["druid"], reason="returns empty string for None", raises=AssertionError
+)
+def test_memtable_null_str_default_dtype(con):
+    """Ensure that null strings are translated to some kind of null value."""
+    t = ibis.memtable(pd.DataFrame({"a": ["a", None, "c"]}))
+    result = con.execute(t.a)
+    assert result.iat[0] == "a"
+    assert pd.isna(result.iat[1])
```
Comment (NickCrews) [inline ibis/backends/tests/test_generic.py:930]: why the general `pd.isna`? What is stopping us from doing `assert result.iat[1] is None`? Adding a comment explaining would be nice

Comment (cpcloud) [reply ibis/backends/tests/test_generic.py:930]: Some backends (it may just be impala, but I don't really want to fix that right now) do not return `None` and return `np.nan`.

File: ibis/expr/datatypes/value.py [hunk 2]
```diff
@@ -267,9 +267,14 @@ def infer_shapely_multipolygon(value) -> dt.MultiPolygon:
 
 def normalize(typ, value):
     """Ensure that the Python type underlying a literal resolves to a single type."""
-    if value is None:
+    import pandas as pd
+
+    if value is None or isinstance(value, type(pd.NA)):
```
Comment (NickCrews) [inline ibis/expr/datatypes/value.py:272]: could this use the same `pd.isna()` check that is used in the test?

File: ibis/tests/strategies.py [hunk 3]
```diff
@@ -187,3 +188,6 @@ def memtable(draw, schema=schema(primitive_dtypes)):  # noqa: B008
 
     df = draw(dataframe)
     return ibis.memtable(df)
+
+
+null_values = st.sampled_from([None, pd.NA])
```
Comment (NickCrews) [inline ibis/tests/strategies.py:193]: Should this include `np.nan`?

Comment (cpcloud) [reply ibis/tests/strategies.py:193]: I don't think so, since that implies we're endorsing `np.nan` as NULL value. While it is used that way in pandas, it's not like that in any other system that isn't based on pandas.

I hope that doesn't sound too ideological. I'm trying to avoid making the mistake of supporting `np.nan` as NULL everywhere and setting expectations according.

File: ibis/expr/datatypes/value.py (hunk 2 above)
Comment (cpcloud) [reply ibis/expr/datatypes/value.py:272]: I don't think so, because I'm intentionally not including `NaN` in this check.

**Comment by cpcloud** [comment]
I don't think this is viable at the moment, closing.
